code/maps/map_prep.R

```diff
@@ -4,79 +4,150 @@ library(readxl)
 ############################
-# merge in census geos
+# clean up base ppp dataset
 ############################
-# clean up zcta to census geo cw
-zcta_mapping <- zcta_mapping %>% 
-    mutate(state_fips_code = str_pad(state_fips_code, 2, pad = "0"),
-           county_fips_code = str_pad(county_fips_code, 3, pad = "0"),
-           tract_fips_code = str_pad(tract_fips_code, 6, pad = "0"),
-           zcta = str_pad(zcta, 5, pad = "0"),
-           zcta_geoid = zcta,
-           state_geoid = state_fips_code,
-           county_geoid = paste0(state_fips_code, county_fips_code),
-           tract_geoid = paste0(state_fips_code, county_fips_code, tract_fips_code),
-           block_group_geoid = paste0(state_fips_code, county_fips_code, tract_fips_code, block_group_fips_code))
-
-# keep only ZCTA-county mapping for now
-zcta_county_mapping <- zcta_mapping %>% 
-    select(zcta_geoid, county_geoid) %>% 
+# add point estimate for each loan
+adbs$LoanAmount<-as.numeric(adbs$LoanAmount)
+adbs <- adbs %>% 
+    mutate(LoanAmount_Estimate_Low = case_when(!is.na(LoanAmount) ~ LoanAmount,
+                                               is.na(LoanAmount) & LoanRange=="a $5-10 million" ~ 5000000,
+                                               is.na(LoanAmount) & LoanRange=="b $2-5 million" ~ 2000000,
+                                               is.na(LoanAmount) & LoanRange=="c $1-2 million" ~ 1000000,
+                                               is.na(LoanAmount) & LoanRange=="d $350,000-1 million" ~ 350000,
+                                               is.na(LoanAmount) & LoanRange=="e $150,000-350,000" ~ 150000),
+           LoanAmount_Estimate_Mid = case_when(!is.na(LoanAmount) ~ LoanAmount,
+                                               is.na(LoanAmount) & LoanRange=="a $5-10 million" ~ 7500000,
+                                               is.na(LoanAmount) & LoanRange=="b $2-5 million" ~ 3500000,
+                                               is.na(LoanAmount) & LoanRange=="c $1-2 million" ~ 1500000,
+                                               is.na(LoanAmount) & LoanRange=="d $350,000-1 million" ~ 675000,
+                                               is.na(LoanAmount) & LoanRange=="e $150,000-350,000" ~ 250000),
+           LoanAmount_Estimate_High = case_when(!is.na(LoanAmount) ~ LoanAmount,
+                                                is.na(LoanAmount) & LoanRange=="a $5-10 million" ~ 10000000,
+                                                is.na(LoanAmount) & LoanRange=="b $2-5 million" ~ 5000000,
+                                                is.na(LoanAmount) & LoanRange=="c $1-2 million" ~ 2000000,
+                                                is.na(LoanAmount) & LoanRange=="d $350,000-1 million" ~ 1000000,
+                                                is.na(LoanAmount) & LoanRange=="e $150,000-350,000" ~ 350000))
+
+# get congressional district code
+state_fips <- states@data[,c('GEOID', 'STUSPS')] %>% 
```

Comment:
I'm sure this is a versioning thing, but I had to change this line to the following to get the flow to work (the "@" syntax caused my version of R (4.0.2) to balk):

state_fips <- states %>%
    rename('state_fips' = 'GEOID',
           'state_name' = 'STUSPS')

Reply:
Ah I see. I get an error if I don't convert `states` to data frame first. I've changed that line to `as.data.frame(states)[,c('GEOID', 'STUSPS')] %>%` which should work for both of us now.